legion/engine/physics/mesh_splitter_utils/mesh_half_edge.hpp

```diff
@@ -0,0 +1,268 @@
+            std::shared_ptr<MeshHalfEdge> third)
+        {
+            first->nextEdge = second;
+            second->nextEdge = third;
+            third->nextEdge = first;
+        }
+ 
+
+        bool IsSplitByPlane(const math::mat4& transform
+            , const math::vec3& planePosition
+            ,const  math::vec3& planeNormal)
+        {
+            int x = 0;
+            int y = 0;
+
+            auto [currentDistFromPlane, nextDistFromPlane] = GetEdgeDistancesFromPlane(transform, planePosition, planeNormal);
+
+            if (currentDistFromPlane > constants::polygonSplitterEpsilon)
+            {
+                x = 1;
+            }
+            else if (currentDistFromPlane < -constants::polygonSplitterEpsilon)
+            {
+                x = -1;
+            }
+
+
+            if (nextDistFromPlane > constants::polygonSplitterEpsilon)
+            {
+                y = 1;
+            }
+            else if (nextDistFromPlane < -constants::polygonSplitterEpsilon)
+            {
+                y = -1;
+            }
+
+            int edgeSplitState = x * y;
+
+
+            return edgeSplitState < 0;
+
+
+        }
+
+        math::vec3 GetWorldCentroid(const math::mat4& transform) const
+        {
+            return transform * math::vec4((position + nextEdge->position) * 0.5f, 1);
+        }
+
+        bool isEdgePartlyAbovePlane(const math::mat4& transform, const math::vec3& planePosition, const math::vec3& planeNormal)
+        {
+            auto [currentDistFromPlane, nextDistFromPlane] = GetEdgeDistancesFromPlane(transform, planePosition, planeNormal);
+
+            bool currentAbovePlane = currentDistFromPlane > constants::polygonSplitterEpsilon;
+            bool nextAbovePlane = nextDistFromPlane > constants::polygonSplitterEpsilon;
+
+            return currentAbovePlane || nextAbovePlane;
+        }
+
+        bool isEdgePartlyBelowPlane(const math::mat4& transform, const math::vec3& planePosition, const math::vec3& planeNormal)
+        {
+            auto [currentDistFromPlane, nextDistFromPlane] = GetEdgeDistancesFromPlane(transform, planePosition, planeNormal);
+
+            bool currentBelowPlane = currentDistFromPlane < -constants::polygonSplitterEpsilon;
+            bool nextBelowPlane = nextDistFromPlane < -constants::polygonSplitterEpsilon;
+
+            return currentBelowPlane || nextBelowPlane;
+        }
+
+        std::tuple<float, float> GetEdgeDistancesFromPlane(const math::mat4& transform, const math::vec3& planePosition, const math::vec3& planeNormal)
+        {
+            auto [currentWorldPos, nextWorldPos] = GetEdgeWorldPositions(transform);
+
+            float currentDistFromPlane = PhysicsStatics::PointDistanceToPlane(planeNormal, planePosition,  currentWorldPos);
+            float nextDistFromPlane = PhysicsStatics::PointDistanceToPlane(planeNormal, planePosition,  nextWorldPos);
+
+            return std::make_tuple(currentDistFromPlane, nextDistFromPlane);
+        }
```

Comment:
which transforms are supposed to be passed in, please comment that (or rather the entire function). Is it the transform matrix of the place or of the edge